### PR TITLE
handle missing UPSTREAM_WORKSPACE file

### LINKED ISSUE #139: Missing rosinstall file is not handled
If `UPSTREAM_WORKSPACE=file` and  `$ROSINSTALL_FILENAME[.$ROS_DISTRO]` does not exist, the build should fail.
@ipa-fxm: FYI

### PR DESCRIPTION
**Pull request by ipa-mdl**
fixes #139

### CODE REVIEW

File: industrial_ci/src/tests/source_tests.sh [hunk 1]
```diff
@@ -78,6 +78,8 @@ file) # When UPSTREAM_WORKSPACE is file, the dependended packages that need to b
     elif [ -e $TARGET_REPO_PATH/$ROSINSTALL_FILENAME ]; then
         # install (maybe unreleased version) dependencies from source
         $ROSWS merge file://$TARGET_REPO_PATH/$ROSINSTALL_FILENAME
+    else
+        error "UPSTREAM_WORKSPACE file '$TARGET_REPO_PATH/$ROSINSTALL_FILENAME[.$ROS_DISTRO]' does not exist"
```
Comment (130s) [inline industrial_ci/src/tests/source_tests.sh]: Output of this line looks a bit confusing:

https://travis-ci.org/ros-industrial/industrial_ci/jobs/208837309#L163
```
\e[31mUPSTREAM_WORKSPACE file '/root//home/travis/build/ros-industrial/industrial_ci/.i.do.not.exist[.jade]' does not exist\e0m
```

- There may be a bug in `error` function. The extraneous prefix in the beginning (`\e[31mUPSTREAM_WORKSPACE`).
- What's the purpose for enclosing the file extension as `[.$ROS_DISTRO]`?

Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]: > What's the purpose for enclosing the file extension as [.$ROS_DISTRO]?

At this point we can't tell which file is missing, the generic or the distro-specific one.

Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]: > There may be a bug in error function.

make it two ;)

**Comment by ipa-mdl** [comment]
The `TARGET_REPO_PATH` is fixed #136

File: industrial_ci/src/tests/source_tests.sh (hunk 1 above)
Comment (130s) [inline industrial_ci/src/tests/source_tests.sh]: >>    What's the purpose for enclosing the file extension as [.$ROS_DISTRO]?
>
> At this point we can't tell which file is missing, the generic or the distro-specific one.

I think I got your point; either generic or distro-specific file is missing but there's no way for the script to know which one is intended by the user.

The `[]` notation may not look the best appropriate for this case, but then I don't know what's the best. Let's just leave this as is :)

**Pull request merged**